modules/utxo-lib/src/bitgo/Unspent.ts

```diff
@@ -97,11 +101,33 @@ export function toPrevOutput(u: Unspent, network: Network): PrevOutput {
  * @param u
  * @param sequence - sequenceId
  */
-export function addToTransactionBuilder(txb: UtxoTransactionBuilder, u: Unspent, sequence?: number): void {
+export function addToTransactionBuilder<TNumber extends number | bigint>(
+  txb: UtxoTransactionBuilder<TNumber>,
+  u: Unspent<TNumber>,
+  sequence?: number
+): void {
   const { txid, vout, script, value } = toPrevOutput(u, txb.network as Network);
   txb.addInput(txid, vout, sequence, script, value);
 }
 
-export function unspentSum(unspents: Unspent[]): number {
-  return unspents.reduce((sum, u) => sum + u.value, 0);
+/**
+ * Sum the values of the unspents.
+ * Throws error if sum is not a safe integer value, or if unspent amount types do not match `amountType`
+ * @param unspents - array of unspents to sum
+ * @param amountType - expected value type of unspents
+ * @return unspentSum - type matches amountType
+ */
+export function unspentSum<TNumber extends number | bigint>(
+  unspents: Unspent<TNumber>[],
+  amountType: 'number' | 'bigint' = 'number'
```

Comment:
New required field in parameters. Please record as BREAKING CHANGE in commits.

Reply:
Isn't this optional? It has a default value.

Reply:
Ope, yup. Ignore those ones

Reply:
Ope, yup. Ignore those.